pyprocessing/math.py

```diff
@@ -162,191 +179,214 @@ def from_angle(angle, target=None):
-        if self.is_2d:
-            return f'{self.x} {self.y}'
-        return f'{self.x} {self.y} {self.z}'
+        return ' '.join(str(i) for i in self)
 
     def __repr__(self):
-        return f'PVector<{", ".join(str(self).split())}> at {id(self)}'
+        return f"PVector<{', '.join(str(i) for i in self)}> at {id(self)}"
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __iter__(self):
+        return iter(self.seq)
 
     def __eq__(self, other):
-        if isinstance(other, PVector):
-            return all(
-                m1 == m2 for m1, m2 in zip(self.position, other.position)
-            )
-        if isinstance(other, (list, tuple)):
-            return all(m1 == m2 for m1, m2 in zip(self.position, other))
-        raise ValueError(
-            f"Invalid operand '==' for type Pvector and {type(other)}"
-        )
+        if not isinstance(other, Iterable):
+            return False
+
+        if not len(other) == len(self):
+            return False
+
+        if not all(component_a == component_b for
+                   component_a, component_b in zip(self, other)):
+            return False
+
+        return True
 
     def __getitem__(self, item):
-        if isinstance(item, str):
-            return {'x': self.x, 'y': self.y, 'z': self.z}[item]
-        if isinstance(item, int):
-            return [self.x, self.y, self.z][item]
-        raise KeyError(f"Item '{item}' cannot be accessed.")
+        return self.seq[item]
+
+    def __setitem__(self, item, v):
+        self.seq[item] = v
+
+    @property
+    def position(self):
+        return tuple(self)
 
     @property
-    def z(self):
-        if self._z is None:
-            return 0
-        return self._z
+    def is_2d(self):
+        return len(self) == 2
 
-    @z.setter
-    def z(self, value):
-        self._z = value
-        if value is not None:
-            self.is_2d = False
+    @property
+    def is_3d(self):
+        return len(self) == 3
 
     @property
-    def position(self):
-        return self.x, self.y, self.z
+    def is_4d(self):
+        return len(self) == 4
 
     def set(self, *vector, x=None, y=None, z=None):
-        if vector:
-            if isinstance(vector[0], PVector):
-                self.x, self.y, self.z = vector[0].position
-            elif isinstance(vector[0], (list, tuple)):
-                self.x, self.y, *_ = vector[0]
-                if len(vector[0]) == 3:
-                    self.z = vector[0][2]
-                else:
-                    self.z = None
-            elif all(isinstance(p, (int, float)) for p in vector):
-                self.x, self.y, self.z = vector
-            elif isinstance(vector[0], complex):
-                self.x, self.y, self.z = vector[0].real, vector[0].imag, None
-        if x is not None:
-            self.x = x
-        if y is not None:
-            self.y = y
-        if z is not None:
-            self.z = z
+        self.seq = PVector(*vector, x, y, z).seq
```

Comment:
we lose the ability to set a Vector from a complex

Reply:
Now __init__ supports complex.